mfhdf/hdiff/hdiff_list.c

```diff
@@ -1183,7 +1183,7 @@ insert_vs(int32 file_id, int32 ref, /* ref of input VS */
     }
 
     /* ignore reserved HDF groups/vdatas; they are lone ones */
-    if (is_lone == 1 && vdata_class != NULL) {
+    if (is_lone == 1 && vdata_class[0] == '\0') {
```

Comment:
so vdata_class has to be empty and reserved in order to detach the vdata_id?

Reply:
vdata_class is a char array and it's just a check to see if it's empty